internal/provider/resource_time_rotating_test.go

```diff
@@ -66,369 +70,1224 @@ func TestAccTimeRotating_Triggers(t *testing.T) {
 	resourceName := "time_rotating.test"
-	timestamp := time.Now().UTC()
+
+	now := time.Now().UTC()
+	mockClock := fakeclock.NewFakeClock(now)
+	rotationDays := 7
 
 	resource.UnitTest(t, resource.TestCase{
-		ProtoV5ProviderFactories: protoV5ProviderFactories(),
-		CheckDestroy:             nil,
+		CheckDestroy: nil,
 		Steps: []resource.TestStep{
 			{
-				Config: testAccConfigTimeRotatingRotationDays(timestamp.Format(time.RFC3339), 7),
+				ProtoV5ProviderFactories: protoV5ProviderFactoriesTestProvider(mockClock),
+				Config:                   testAccConfigTimeRotatingRotationDays(rotationDays),
+				ConfigPlanChecks: resource.ConfigPlanChecks{
+					PreApply: []plancheck.PlanCheck{
+						plancheck.ExpectResourceAction(resourceName, plancheck.ResourceActionCreate),
+						plancheck.ExpectUnknownValue(resourceName, tfjsonpath.New("rotation_rfc3339")),
+						plancheck.ExpectUnknownValue(resourceName, tfjsonpath.New("rfc3339")),
+						plancheck.ExpectKnownValue(resourceName, tfjsonpath.New("rotation_days"), knownvalue.Int64Exact(int64(rotationDays))),
+					},
+				},
 				ConfigStateChecks: []statecheck.StateCheck{
-					statecheck.ExpectKnownValue(resourceName, tfjsonpath.New("rotation_days"), knownvalue.Int64Exact(7)),
-					statecheck.ExpectKnownValue(resourceName, tfjsonpath.New("rotation_rfc3339"), knownvalue.StringExact(timestamp.AddDate(0, 0, 7).Format(time.RFC3339))),
-					statecheck.ExpectKnownValue(resourceName, tfjsonpath.New("rotation_years"), knownvalue.Null()),
-					statecheck.ExpectKnownValue(resourceName, tfjsonpath.New("rotation_months"), knownvalue.Null()),
-					statecheck.ExpectKnownValue(resourceName, tfjsonpath.New("rotation_hours"), knownvalue.Null()),
-					statecheck.ExpectKnownValue(resourceName, tfjsonpath.New("rotation_minutes"), knownvalue.Null()),
-					statecheck.ExpectKnownValue(resourceName, tfjsonpath.New("rfc3339"), knownvalue.NotNull()),
+					statecheck.ExpectKnownValue(resourceName, tfjsonpath.New("rotation_rfc3339"), knownvalue.StringExact(now.AddDate(0, 0, rotationDays).Format(time.RFC3339))),
+					statecheck.ExpectKnownValue(resourceName, tfjsonpath.New("rfc3339"), knownvalue.StringExact(now.Format(time.RFC3339))),
+					statecheck.ExpectKnownValue(resourceName, tfjsonpath.New("rotation_days"), knownvalue.Int64Exact(int64(rotationDays))),
 				},
 			},
 			{
-				ResourceName:      resourceName,
-				ImportState:       true,
-				ImportStateIdFunc: testAccTimeRotatingImportStateIdFunc(),
-				ImportStateVerify: true,
+				ProtoV5ProviderFactories: protoV5ProviderFactoriesTestProvider(mockClock),
+				ResourceName:             resourceName,
+				ImportState:              true,
+				ImportStateIdFunc:        testAccTimeRotatingImportStateIdFunc(),
+				ImportStateVerify:        true,
+			},
+			// Trigger a rotation
+			{
+				PreConfig: func() {
+					rotationDate := now.AddDate(0, 0, -8)
+					mockClock.Increment(mockClock.Since(rotationDate)) // 8 days
```

Comment:
nit: This was a little confusing to read initially. If we implement a fake clock maybe we can just implement a passthrough to the `AddDate` method